hooks/useLocal.ts

```
@@ -133,6 +133,19 @@ export const useLocal = () => {
       throw new Error(ErrorConstants.FAILED_TO_SAVE_PATH_PROGRESS);
     }
   };
+  const handleUpdatePathWithErrorHandling = async (
```

Why do we need a separate function for try catch? Can't we do that directly in handleUpdatePath() function?

---

Initially, we handled errors directly inside handleUpdatePath(), but it caused alerts to show even when the debounced update failed. So we removed the internal try–catch and let it throw errors instead.
Since we needed UI feedback (alerts) in multiple places, I created a reusable try–catch wrapper to handle those cases consistently.